hotel_housekeeping_planning/models/hotel_room.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2020 Coop IT Easy SCRL fs
+#   Robin Keunen <robin@coopiteasy.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from datetime import date, timedelta
+
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+from odoo.fields import Date, Datetime
+
+
+class HotelRoom(models.Model):
+    _inherit = "hotel.room"
+
+    @api.multi
+    def action_housekeeping_planning_report(self):
+        rooms = self.search([])
+        return self.env.ref(
+            "hotel_housekeeping_planning"
+            ".action_hotel_housekeeping_planning_report"
+        ).report_action(rooms)
+
+    @api.multi
+    def get_occupation(self, day):
+        """day: date object"""
+        assert type(day) == date, "day should be of type datetime.date"
+        self.ensure_one()
+
+        reservation_lines = self.env["hotel_reservation.line"].search(
+            [
+                ("line_id.state", "in", ["confirm", "done"]),
+                ("reserve", "=", self.id),
+                # checkin during the day or sooner
+                ("checkin", "<=", Date.to_string(day)),
+                # checkout during the day or later
+                ("checkout", ">=", Date.to_string(day)),
+            ]
+        )
+
+        if len(reservation_lines) >= 2:
+            return "departure_arrival"
+        elif len(reservation_lines) == 0:
+            return "free"
+        elif Datetime.from_string(reservation_lines.checkin).date() == day:
+            return "arrival"
+        elif Datetime.from_string(reservation_lines.checkout).date() == day:
+            return "departure"
+        else:
+            return "busy"
+
+    @api.model
+    def _get_week(self):
+        today = date.today()
+        monday = today - timedelta(days=today.weekday())
+        return [monday + timedelta(days=i) for i in range(7)]
+
+    @api.multi
+    def _get_notes(self):
+        self.ensure_one()
+        today = date.today()
+        monday = today - timedelta(days=today.weekday())
+        sunday = monday + timedelta(days=6)
+
+        reservation_lines = (
+            self.env["hotel_reservation.line"]
+            .search(
+                [
+                    ("line_id.state", "in", ["confirm", "done"]),
+                    ("reserve", "=", self.id),
+                    # checkin sunday or sooner
+                    ("checkin", "<=", Date.to_string(sunday)),
+                    # checkout monday or later
+                    ("checkout", ">=", Date.to_string(monday)),
+                ]
+            )
+            .sorted(lambda rl: rl.checkin)
+        )
+        notes = (rl.line_id.housekeeping_note for rl in reservation_lines)
+        notes = (note for note in notes if note)  # remove empty notes (False)
+        return "\n".join(notes)
```

Comment:
Could it be an idea, for clarity, to include a reference to the reservation (and it's rooms) from which each note originates?
```

Reply:
I guess you're right, I'm afraid it'd be too long in the report but I'll do it anyway.